src/gcs-resumable-upload/index.ts

```diff
@@ -1053,14 +1016,11 @@ export class Upload extends Pumpify {
    */
   private onResponse(resp: GaxiosResponse) {
     if (
-      (this.retryableErrorFn &&
-        this.retryableErrorFn({
-          code: resp.status,
-          message: resp.statusText,
-          name: resp.statusText,
-        })) ||
-      resp.status === NOT_FOUND_STATUS_CODE ||
```

Comment:
Tracking this down, this looks like a precursor to our retry efforts. I’m also not too sure why it’s 404 in particular

Reply:
It appears the service can return `404` if it's been more than a week since the upload was initiated https://cloud.google.com/storage/docs/resumable-uploads#session-uris

Reply:
I see. In this case, the error is retryable because retrying will initiate a new upload?

Reply:
I think we would be technically correct and conformant to not retry 404s automatically in this case